examples/state_management_example/README.md

```diff
@@ -0,0 +1,160 @@
+
+```typescript
+class CounterActions extends BaseActions<CounterState> {
+  // Works with both Observable and Redux backends
+  constructor(adapterOrObservable: StateAdapter<CounterState> | ObservableLike<CounterState>) {
+    super(adapterOrObservable);
+  }
+
+  increment() {
+    this.updateState(state => ({
+      ...state,
+      count: state.count + 1
+    }));
+  }
+}
+```
+
+### BaseSelectors
+
+Access state in a read-only manner:
+
+```typescript
+class CounterSelectors extends BaseSelectors<CounterState> {
+  // Works with both Observable and Redux backends
+  constructor(adapterOrObservable: StateAdapter<CounterState> | ObservableLike<CounterState>) {
+    super(adapterOrObservable);
+  }
+  
+  // The base class provides getState() by default
+  isPositive() {
+    return this.getState().count > 0;
+  }
+}
+```
+
+## React Hooks
+
+### useAction
+
+Access actions from a registered service:
+
+```typescript
+// In a React component
+const actions = useAction<CounterState>('counterService');
+// Then use actions
+<button onClick={() => actions.increment()}>+</button>
+```
+
+### useSelector
+
+Subscribe to state changes:
+
+```typescript
+// Only re-renders when the selected value changes
+const count = useSelector<CounterState, number>(
+  'counterService',
+  state => state.count
+);
+```
+
+## Example Implementations
+
+### Observable-Based Counter
+
+```typescript
+// Create with Observable pattern
+const state$ = new BehaviorSubject<CounterState>({ count: 0 });
+const actions = new CounterActions(state$);
+const selectors = new CounterSelectors(state$);
+
+// Register with global store
+globalStoreServiceRegister('observableCounter', selectors, actions);
+```
+
+### Redux-Based Counter
+
+```typescript
+// Create with Redux pattern
+const store = configureStore({
+  reducer: { counter: counterReducer }
+});
+const adapter = new ReduxStateAdapter<CounterState>(store, state => state.counter);
+const actions = new CounterActions(adapter);
+const selectors = new CounterSelectors(adapter);
+
+// Register with global store
+globalStoreServiceRegister('reduxCounter', selectors, actions);
+```
+
+### Component Usage
+
+```tsx
+// The same component works with both backends
+function CounterDisplay({ serviceKey }) {
+  const count = useSelector(serviceKey, state => state.count);
+  const actions = useAction(serviceKey);
+  
+  return (
+    <div>
+      <p>Count: {count}</p>
+      <button onClick={() => actions.increment()}>Increment</button>
+    </div>
+  );
+}
+
+// Use it with either backend
+<CounterDisplay serviceKey="observableCounter" />
+<CounterDisplay serviceKey="reduxCounter" />
+```
+
+## Key Benefits
+
+1. **Backend Flexibility**: Choose the state management pattern that best fits your needs
+2. **Consistent API**: Components use the same hooks regardless of backend
+3. **Easy Migration**: Switch backends without changing component code
+4. **Type Safety**: Full TypeScript support throughout
+
```

Comment:
another benefit is easier to contribute state across plugins if there's a global registry